ksqldb-engine/src/main/java/io/confluent/ksql/util/SharedKafkaStreamsRuntimeImpl.java

```diff
@@ -176,7 +176,7 @@ public TimeBoundedQueue getNewQueryErrorQueue() {
   public void stop(final QueryId queryId, final boolean resetOffsets) {
     log.info("Attempting to stop query: {} in runtime {} with resetOffsets={}",
              queryId, getApplicationId(), resetOffsets);
-    if (collocatedQueries.containsKey(queryId)) {
+    if (kafkaStreams.getTopologyByName(queryId.toString()).isPresent()) {
```

Comment:
Why we cannot use the old condition as well, and why not keep the `remove` call below inside the if condition?

Reply:
If the query was registered but not started this causes issues when removing form streams. This was just the simplest way I could get it to work

Reply:
Got it. My understanding is that:

1) If the query is registered, it would be in the `collocatedQueries`.
2) If the query is started, it would be added to `kafkaStreams`.
3) The query must always be registered before started, i.e. 2) always happens after 1).

So upon `stop`, the query can either be registered but not started, or started. Is that right? My question was that could it ever be not registered at all (i.e. not in `collocatedQueries`)? If not, then this fix looks elegant.